### PR TITLE
Specify a minimum supported version for VxWorks

### PR DESCRIPTION
**Pull request by devnexen**
Document that VxWorks 7 is the minimum supported version.

### CODE REVIEW

**Comment by devnexen** [comment]
r? tgross35

**Comment by tgross35** [comment]
This PR seeks to specify a minimum supported version of VxWorks to 7 to make it more obvious which APIs should be considered usable. Does this seem reasonable, or should a different version be specified?

Asking some people who have contributed to vxworks here or in libc, please comment or thumbs up if the proposed version seems good. @biabbas @jdygert-spok @NicholasBaron @BaoshanPang

Some more context is at https://github.com/rust-lang/rust/pull/128758

(also, it would be good to have one or two more listed maintainers for VxWorks, which just means you can get pinged when platform-specific questions come up. If anyone here is interested, just submit a PR updating `src/doc/rustc/src/platform-support/vxworks.md`.)

File: src/doc/rustc/src/platform-support/vxworks.md
Comment (tgross35) [inline src/doc/rustc/src/platform-support/vxworks.md:25]: Why are the platforms specified - is something different for ppc and arm32?

**Comment by tgross35** [comment]
Just to clarify this should get some feedback before merge:

@rustbot blocked

File: src/doc/rustc/src/platform-support/vxworks.md
Comment (biabbas) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: Its the same for all targets. 

Note: 
#128545

Comment (tgross35) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: Thanks for confirming. @devnexen could you drop the "on x86/64 and ARM64" unless there is a specific reason?

**Comment by biabbas** [comment]
> This PR seeks to specify a minimum supported version of VxWorks to 7 to make it more obvious which APIs should be considered usable. Does this seem reasonable, or should a different version be specified?
> 
> Asking some people who have contributed to vxworks here or in libc, please comment or thumbs up if the proposed version seems good: @biabbas @jdygert-spok @NicholasBaron @BaoshanPang
> 
> Some more context is at #128758
> 
> (also, it would be good to have one or two more listed maintainers for VxWorks, which just means you can get pinged when platform-specific questions come up. If anyone here is interested, just submit a PR updating `src/doc/rustc/src/platform-support/vxworks.md`.)

I'm not sure if anyone else is also working on VxWorks rust, thus no one else is listed as a maintainer.

File: src/doc/rustc/src/platform-support/vxworks.md
Comment (tgross35) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: @biabbas does this minimum version seem reasonable to you?

Also, could someone confirm when this version (VxWorks 7) was released? I am having some trouble figuring this out online what the oldest supported version will be.

Comment (biabbas) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: I think VxWorks 7 was released in 2016. Rust was enable for VxWorks in 2019(#61946 ). I've only worked on VxWorks 7.

Comment (biabbas) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: I think Versions older than VxWorks7 would require porting efforts, minimum supported version should be VxWorks7.

Comment (biabbas) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: I think this pr would cause merge conflicts with #128545.

Comment (biabbas) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: > I think Versions older than VxWorks7 would require porting efforts, minimum supported version should be VxWorks7.
But since we would have to go through the changes from VxWorks 6 to VxWorks 7 to confirm the minimum supported version. This seems less reasonable now.

Comment (devnexen) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: I think your PR should be merged first.

Comment (biabbas) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: @devnexen 
Do you have access to VxWorks 6 workbench?

Comment (devnexen) [reply src/doc/rustc/src/platform-support/vxworks.md:25]: nope.

Comment (biabbas) [reply src/doc/rustc/src/platform-support/vxworks.md]: > Thanks for confirming. @devnexen could you drop the "on x86/64 and ARM64" unless there is a specific reason?
After this, it seems ok for now.

**Review by biabbas — APPROVED** [review]

File: src/doc/rustc/src/platform-support/vxworks.md
Comment (tgross35) [inline src/doc/rustc/src/platform-support/vxworks.md:25]:
```suggestion
The minimum supported version is VxWorks 7.
```

**Comment by tgross35** [comment]
If the specified version is from 2016, I think it is pretty unobjectionable. That is pretty similar to our oldest MacOS (10.12, 2016) and Windows (10, 2015). If there are any problems we can revisit, but this should make the ongoing `std` support effort more straightforward.

Thanks for getting this documented.

@bors r+ rollup

**Pull request merged**
